packages/chat-stateful-client/src/ChatClientState.ts

```diff
@@ -45,3 +51,61 @@ export type ChatThreadClientState = {
 export type ChatThreadProperties = {
   topic?: string;
 };
+
+/**
+ * Errors teed from API calls to the Chat SDK.
+ *
+ * Each property in the object stores the latest error for a particular SDK API method.
+ *
+ * Errors from this object can be cleared by calling the TODO(implement me) {@Link clearError} method.
+ * Additionally, errors are automatically cleared when:
+ * - The state is cleared.
+ * - Subsequent calls to related API methods succeed.
+ * See documentation of individual stateful client methods for details on when errors may be automatically cleared.
+ */
+export type ChatErrors = {
+  [target in ChatErrorTargets]: Error;
+};
+
+/**
+ * Error thrown from failed stateful API methods.
+ */
+export class ChatError extends Error {
+  /**
+   * The API method target that failed.
```

Comment:
Maybe we could move all these types to somewhere and make this file shorter?

Reply:
Will look at splitting up this file in one of my follow up PRs.